environment.go

```diff
@@ -930,12 +930,16 @@ func getClientConfig(baseURL string) (*ClientConfig, error) {
 
 		parts := strings.Split(path, string(filepath.Separator))
 		buildInfo := strings.Split(parts[len(parts)-2], "_")
-
+		displayName := ValidArchDisplayNames[fmt.Sprintf("%s_%s", buildInfo[0], buildInfo[1])]
+		if displayName == "" {
+			displayName = parts[len(parts)-2]
```

Comment:
FYI I put this default because I'm annoyed that on prod the legacy linux/amd64 doesn't say legacy.

Reply:
It was intentional wasn't it? I think we wanted to not have people download the legacy one, but if it's there with a different name then we messed something up probably

Reply:
Right now they're both called linux/amd64 which means users might just download either one because they're confused.

Reply:
Right, so shouldn't we filter the legacy one off of the client config, rather than rename it?